app/src/main/java/Fragments/SettingsFragment.java

```diff
@@ -31,17 +36,39 @@ public static SettingsFragment newInstance() {
     public void onCreate(Bundle savedInstanceState)
     {
         super.onCreate(savedInstanceState);
-
-
     }
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_settings, container, false);
+        View view = inflater.inflate(R.layout.fragment_settings, container, false);
+        return view;
     }
 
-
-
+    @Override
+    public void onStart() {
```

Comment:
I will be moving the Authentication State Listener from onStart() to onCreateView() as it's less likely to be called redundantly.